src/ruby/puppetserver-lib/puppet/server/compiler.rb

```diff
@@ -48,11 +81,11 @@ def convert_java_args_to_ruby(hash)
         Hash[hash.collect do |key, value|
           # Stolen and modified from params_to_ruby in handler.rb
           if value.java_kind_of?(Java::ClojureLang::IPersistentMap)
-            [newkey, convert_java_args_to_ruby(value)]
+            [key, convert_java_args_to_ruby(value)]
           elsif value.java_kind_of?(Java::JavaUtil::List)
-            [newkey, value.to_a]
+            [key, value.to_a]
           else
-            [newkey, value]
+            [key, value]
```

Comment:
@tvpartytonight just put up a PR (#1987) to fix this. We can either close that or rebase this to pick that up if we merge it first.

Reply:
Yeah, I'll rebase this out once everything else looks good.